src/main/java/com/example/demo/controller/MovieController.java

```diff
@@ -0,0 +1,169 @@
+package com.example.demo.controller;
+
+import com.example.demo.entity.Booked;
+import com.example.demo.entity.BookedSeat;
+import com.example.demo.entity.Movie;
+import com.example.demo.repository.IMovieRepository;
+import com.example.demo.service.MovieService;
+import lombok.AllArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+
+@RestController("movies")
+@AllArgsConstructor
+public class MovieController {
+
+    final IMovieRepository movieRepository;
+
+    final MovieService movieService;
+
+
+    public static Boolean nameVerification(String name) {
+        String regex = "[A-ZĄĆĘŁŃÓŻŹ][a-ząćęłńóżź][a-ząćęłńóżź]+ " +
+                "[A-ZĄĆĘŁŃÓŻŹ][a-ząćęłńóżź][a-ząćęłńóżź]+(-[A-ZĄĆĘŁŃÓŻŹ][a-ząćęłńóżź][a-ząćęłńóżź]+)?";
+        Pattern pattern = Pattern.compile(regex);
+        if(pattern.matcher(name).matches()) {
+            return Boolean.TRUE;
+        }
+        return Boolean.FALSE;
+    }
+
+    public static Boolean requestValidation(String name, Movie movie) {
+        if(movie.getScreeningDateTime().isAfter(LocalDateTime.now().plusMinutes(15))) {
+            if (nameVerification(name).equals(Boolean.TRUE)) {
+                return Boolean.TRUE;
+            }
+            return Boolean.FALSE;
+        }
+        return Boolean.FALSE;
+    }
+
+
+    @PostMapping
+    public ResponseEntity<Movie> save(@RequestBody Movie movie) {
+        try {
+            return new ResponseEntity<>(movieRepository.save(movie), HttpStatus.CREATED);
+        } catch (Exception e) {
+            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
```

Comment:
czy na pewno to jest kod który powinna zwrócić aplikacja żeby klient wiedział co się stało?